### PR TITLE
Build: Remove `lerna`

### PR DESCRIPTION
**Pull request by valentinpalkovic**
Closes N/A

## What I did

Remove lerna as a dependency since it is not needed anymore.

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

**Review by yannbf — CHANGES_REQUESTED** [review]
Lerna is still needed in certain places. If you search for its usage in .js or .sh files, you will see that it's needed in:
- `generate-sb-packages-versions.js`: used to generate the versions.ts file
- `build-package.js`: used as `yarn build <storybook-package-name>`
- `publish.sh`: I don't know if we even use this, but it does need lerna

**Comment by JReinhold** [comment]
I would also assume that all those scripts still referencing `lerna` should be deleted.

**Comment by valentinpalkovic** [comment]
Good catch @yannbf! Applied the requested changes

File: scripts/build-package.js [hunk 1]
```diff
@@ -4,26 +4,21 @@
...
-  }, []);
-};
+async function getWorkspaces(includePrivate = true) {
+  const { execaCommand } = await import('execa');
+  const { stdout } = await execaCommand(`yarn workspaces list --json --no-private`, {
+    cwd: join(__dirname, '..', 'code'),
+    shell: true,
+  });
+  return JSON.parse(`[${stdout.split('\n').join(',')}]`);
+}
```
Comment (valentinpalkovic) [inline scripts/build-package.js:14]: Almost the same as the function in `scripts/utils/workspace.ts`. But this is a Javascript file, from where I cannot import Typescript files I guess.

Comment (ndelangen) [reply scripts/build-package.js:14]: convert it to TS?

File: scripts/utils/workspace.ts [hunk 2]
```diff
@@ -4,11 +4,14 @@ import { execaCommand } from './exec';
...
 export type Workspace = { name: string; location: string };
 
-export async function getWorkspaces() {
-  const { stdout } = await execaCommand('yarn workspaces list --json', {
-    cwd: CODE_DIRECTORY,
-    shell: true,
-  });
+export async function getWorkspaces(includePrivate = true) {
+  const { stdout } = await execaCommand(
+    `yarn workspaces list --json ${includePrivate ? '' : '--no-private'}`,
```
Comment (ndelangen) [inline scripts/utils/workspace.ts:9]: NICE!, no need to manually filter.

**Comment by ndelangen** [comment]
Does this need any manual steps to test?
Maybe test the scripts @yannbf mentioned?

File: scripts/build-package.js (hunk 1 above)
Comment (valentinpalkovic) [reply scripts/build-package.js:14]: node doesn't execute ts, does it? So I have to use ts-node instead to run the script?

**Comment by valentinpalkovic** [comment]
- `generate-sb-packages-versions.js` is not in use anymore. `scripts/release/version.ts` will bump the versions in the generated `versions.ts` file
- `publish.sh` is not in use anymore

**Comment by valentinpalkovic** [comment]
@ndelangen, @yannbf Applied the requested changes!

**Review by ndelangen — APPROVED** [review]

**Review by yannbf — CHANGES_REQUESTED** [review]
The `yarn build <package>` command now does not give any output on success, and just crashes on error, instead of providing the correct error message like before. Can you fix that?

**Comment by valentinpalkovic** [comment]
@yannbf Fixed!

**Review by yannbf — APPROVED** [review]
Awesome <3

**Pull request merged**